File: 0378-kth-smallest-element-in-a-sorted-matrix/0378-kth-smallest-element-in-a-sorted-matrix.cpp
class Solution {
public:
    void insort(vector<int>& aInOutVec, int aInValue) {
        auto lIt = lower_bound(aInOutVec.begin(), aInOutVec.end(), aInValue);
        aInOutVec.insert(lIt, aInValue);
    }

    int kthSmallest(vector<vector<int>>& matrix, int k) {
        vector<int> lVec;
        for (int i = 0; i < matrix.size(); i++)
        {
            for (int j = 0; j < matrix[i].size(); j++)
            {
                insort(lVec, matrix[i][j]);
            }
        }

        return lVec[k - 1];
    }
};

File: 0378-kth-smallest-element-in-a-sorted-matrix/0378-kth-smallest-element-in-a-sorted-matrix.py
class Solution:
    def kthSmallest(self, matrix: List[List[int]], k: int) -> int:

        L = []
        heapq.heapify(L)
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                heapq.heappush(L, matrix[i][j])
        
        for _ in range(k - 1):
            heapq.heappop(L)

        return heapq.heappop(L)